### PR TITLE
chore: Migrating to melos >= 3

### PR DESCRIPTION
**Pull request by erickzanardo**
Small PR to migrate the project structure to the one needed to use on melos >=3

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -23,4 +23,6 @@ pubspec.lock
 .settings
 .last_build_id
 
-service-account.json
\ No newline at end of file
+service-account.json
+
+pubspec_overrides.yaml
```
Comment (spydon) [inline .gitignore:28]:
```suggestion
**/pubspec_overrides.yaml
```
It should ignore all pubspec overrides files right?

**Comment by rrousselGit** [comment]
I don't want to use v3 or above in my projects, because I don't want to have to define a root-level pubspec.

**Comment by erickzanardo** [comment]
> I don't want to use v3 or above in my projects, because I don't want to have to define a root-level pubspec.

But given that that is the path that melos is going, shouldn't we just keep up with it? I made this first PR since I am planing on sending a future contribution and I have multiple projects using melos, I feel it is more troublesome to keep changing my melos install per project than to have a top level pubspec.

Unless there is a way of keeping multiple versions of melos in my computer that I am unfamiliar wiht?

**Comment by rrousselGit** [comment]
I'd rather not use melos than having to specify a root `pubspec.yaml`.

If you want to contribute something here, feel free to remove melos entirely (in favour if plain pubspec_overrides.yaml)

**Comment by spydon** [comment]
> I'd rather not use melos than having to specify a root `pubspec.yaml`.
> 
> If you want to contribute something here, feel free to remove melos entirely (in favour if plain pubspec_overrides.yaml)

Now I'm curious, what downsides do you see with having a root-level pubspec?

**Comment by rrousselGit** [comment]
It's invasive. I want projects to be as clean as possible from config files.

Dart is really good at having very little config. I want to keep it this way.

**Comment by Salakar** [comment]
We should **not** be gating contributions over personal preference here, the standard on most Invertase repositories is to use Melos and the root pubspec layout, it has served us well at scale when there's a focus on multiple contributors, rather than ones own preferences.

File: pubspec.yaml [hunk 2]
```diff
@@ -0,0 +1,6 @@
+name: dart_firebase_admin_workspace
+
```
Comment (Salakar) [inline pubspec.yaml:2]: Can we add publish to none here?

Comment (erickzanardo) [reply pubspec.yaml:3]: For sure. Added!

**Review by Salakar — APPROVED** [review]

**Pull request merged**